### PR TITLE
SN-FRONT-30 Add socket.io

### PR DESCRIPTION
**Pull request by Aretemal**

### CODE REVIEW

File: src/store/store.ts [hunk 1]
```diff
@@ -22,20 +22,19 @@ const persistConfig = {
   key: 'root',
   storage,
 };
+const persistedReducer = persistReducer(persistConfig, authReducer);
 
 const reducers = combineReducers({
-  auth: authReducer,
+  auth: persistedReducer,
```
Comment (Konstantsin) [inline src/store/store.ts:28]: I would say `persistenceReducer`

File: src/components/Dialogs/Messages/MessagesContainer.tsx [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
+  }, []);
 
   const onSendMessage = (message: string, id: string) => {
-    dispatch(sendMessage({ token, message, id }));
+    dispatch(sendMessage({
+      token, message, id, socket,
+    }));
+  };
+  const onBack = () => {
+    socket.emit('DIALOG:LEAVE_FROM_DIALOG', { dialogId: activeId, userId: authId });
```
Comment (Konstantsin) [inline src/components/Dialogs/Messages/MessagesContainer.tsx:35]: `DIALOG:LEAVE_FROM_DIALOG` should be `DIALOG:LEAVE_DIALOG`

File: src/components/Dialogs/Messages/MessagesContainer.tsx [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
+    dispatch(setDialogId(''));
```
Comment (Konstantsin) [inline src/components/Dialogs/Messages/MessagesContainer.tsx:36]: Does `setDialogId` means `setActiveDialog`?

File: src/store/slices/interfaces/dialogInterface.ts [hunk 4]
```diff
@@ -20,11 +20,14 @@ export interface IDialog {
...
+    dialogId: string,
+    socket: any,
+    userId: string,
 }
 
 export interface ISendMessage{
     token: string,
     id: string,
     message: string,
+    socket: any,
```
Comment (Konstantsin) [inline src/store/slices/interfaces/dialogInterface.ts:32]: Doesn't socket have an interface or type?

File: src/store/slices/dialogSlice.ts [hunk 5]
```diff
@@ -40,15 +44,15 @@ export default dialogSlice.reducer;
...
+  async (data) => {
+    await dialogAPI.sendMessage({ id: data.id, token: data.token, message: data.message });
   },
 );
 export const getAllMessages = createAsyncThunk<void, IGetAllMessages, { dispatch: AppDispatch }>(
   'dialog/getAllMessage',
   async (data, { dispatch }) => {
-    const response = await dialogAPI.getAllMessages(data);
+    const response = await dialogAPI.getAllMessages({ token: data.token, id: data.dialogId });
+    data.socket.emit('DIALOG:JOIN_TO_DIALOG', { dialogId: data.dialogId, usersId: data.userId });
```
Comment (Konstantsin) [inline src/store/slices/dialogSlice.ts:55]: JOIN_DIALOG

File: src/store/store.ts (hunk 1 above)
Comment (Aretemal) [reply src/store/store.ts:28]: done

File: src/components/Dialogs/Messages/MessagesContainer.tsx (hunk 2 above)
Comment (Aretemal) [reply src/components/Dialogs/Messages/MessagesContainer.tsx:35]: done

File: src/components/Dialogs/Messages/MessagesContainer.tsx (hunk 3 above)
Comment (Aretemal) [reply src/components/Dialogs/Messages/MessagesContainer.tsx:36]: done

File: src/store/slices/interfaces/dialogInterface.ts (hunk 4 above)
Comment (Aretemal) [reply src/store/slices/interfaces/dialogInterface.ts:32]: done

File: src/store/slices/dialogSlice.ts (hunk 5 above)
Comment (Aretemal) [reply src/store/slices/dialogSlice.ts:55]: done

**Pull request merged**